receiver/prometheusexec/subprocessmanager/manager.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package subprocessmanager
```

Comment:
Does `subprocessmanager` need to be in a separate package? It's not that large; I would recommend just putting it directly in the `prometheusexecreceiver` package, unless you have a good reason why you want it to be a separate package such that people can consume it as a standalone package.

Reply:
I had discussions in the past with Quentin where we thought it would be a good idea to leave it as a separate package other can consume! It keeps track of runtime and has methods to compute exponential backoff, pipe the output of the subprocesses and format the environment variables!